src/main/java/com/github/grignaak/collections/CowCollectors.java

```diff
@@ -0,0 +1,17 @@
+package com.github.grignaak.collections;
+
+import java.util.List;
+import java.util.stream.Collector;
+
+public final class CowCollectors {
+    public static <T> Collector<T, ?, List<T>> toList() {
+        return new ToListCollector<>();
+    }
+
+    public static <T> Collector<T, ?, List<T>> toForkedList() {
+        return new ToForkedListCollector<>();
```

Comment:
This can be simplified to `Collectors.collectingAndThen(toList(), CowList::fork)`.

**But** I've been thinking about this and I think this method is not only unnecessary, but is actually detrimental (sorry about putting that into the original issue).

The fork method only benefits if there is a reference to the original as well as the forked copy, and collectors don't expose the original collection. But then it has frozen the underlying structure anyway, making subsequent updates a bit more costly.

Overall, I think the better pattern would be to just use `toList` and let the user fork later.